controllers/oauth.go

```diff
@@ -24,13 +24,30 @@ type OAuthResBodyFields struct {
 	// `mentor` or `student`
 	Type string `json:"type"`
 	// Whether the user has newly registered or was registered before
-	IsNewUser bool   `json:"is_new_user"`
-	Jwt       string `json:"jwt"`
+	IsNewUser bool `json:"is_new_user"`
+	// The generated JWT string for the user
+	Jwt string `json:"jwt"`
 }
 
 const OAUTH_TYPE_STUDENT string = "student"
 const OAUTH_TYPE_MENTOR string = "mentor"
 
+// OAuth godoc
+//
+//	@Summary		OAuth Authentication
+//	@Description	Authenticates a user throught Github OAuth.
+//	@Accept			json
+//	@Produce		json
+//	@Param			request	body		OAuthReqBodyFields	true	"Fields required for authentication."
+//	@Success		200		{object}	OAuthResBodyFields	"Succesfully authenticated."
+//	@Failure		400		{object}	utils.HTTPMessage	"Error decoding JSON body."
+//	@Failure		400		{object}	utils.HTTPMessage	"Empty body parameters."
+//	@Failure		500		{object}	utils.HTTPMessage	"Error getting OAuth access token."
```

Comment:
only the last `@Failure  500` annotation is considered

Reply:
Is it not possible to add multiple?

Reply:
The error code is used as a key, so no

Reply:
I can see two options here
- document this in the description tag of the API documenting all the possible behaviours (btw you can add [description over multiple lines](https://github.com/swaggo/swag#descriptions-over-multiple-lines))
- Or, write a long description for the failure tag .

Reply:
hmm

Reply:
I'll choose the second option

Reply:
Done, please review.